text/trace/0107-sampling-priority.md

```diff
@@ -0,0 +1,79 @@
+# Associating sampling priority with the trace using tracestate
+
+Propagating `sampling.priority` (calculated by probability sampler) in a
+`tracestate` will enable a better experience for system with the independent
+stateless configuration of components sampling.
+
+## Motivation
+
+Consistent sampling decision made in each app of a distributed trace is
+important for better user experience of traces analysis. Consistency is achieved
+by aligning stateless hashing algorithm used to make a decision based on
+trace-id or explicitly propagating sampling flag.
+
+Increasingly more apps participate in distributed tracing. With the
+standardized wire format for context propagation, there is bigger chance that
+approach for sampling chosen by one app may not match another app sampling
+policy. Since coordination of sampling policies across many apps not always
+possible, OpenTelemetry must provide a way to share sampling hints between apps.
+
+Beyond sampling hints infrastructure, it is beneficial for the community to
+agree on standard behavior and hints exposed by probability sampler.
+
+The `sampling.priority` property is used by many vendors for various purposes.
+Propagating this field alongside the trace will allow for many improvements and
+as a very minimum will simplify transition of customers from SDKs using
+different `trace-id` based hash functions to OpenTelemetry SDK.
+
+Also see discussion here: <https://github.com/open-telemetry/opentelemetry-specification/pull/570>
+
+## Explanation
+
+### Sampling hints exchange
+
+OpenTelemetry SDK has an infra to expose sampling hints as span attributes today
+via
+[SamplingResult](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#shouldsample)
+class.
+
+This recommendation suggests OpenTelemetry will allow to return an updated
+`tracestate` that will be used for a newly created `Span`. This will allow
+samplers from a single vendor coordinate sampling decisions across the
+different components of the trace.
+
+### Probability Sampler default behavior
+
+The default behavior of probability sampler is to calculate a stateless hash
```

Comment:
Note that this is not yet agreed upon, although there are some PRs open namely https://github.com/open-telemetry/opentelemetry-specification/pull/331 which is several months old and just yesterday we got https://github.com/open-telemetry/opentelemetry-specification/pull/611

Reply:
Is this behavior even relevant to this OTEP? I think _whichever_ algorithm is used by the sampler, as long as it can express the result via a "probability" value (not every sampler can, e.g. rate limiting), the actual calculation is not relevant.